aten/src/ATen/native/Bucketization.cpp

```diff
@@ -89,14 +89,21 @@ void dispatch(Tensor& result, const Tensor& input, const Tensor& boundaries, boo
 
 Tensor& searchsorted_out_cpu(const Tensor& sorted_sequence, const Tensor& self, bool out_int32, bool right, Tensor& result) {
   searchsorted_pre_check(sorted_sequence, self, result, out_int32);
+  Tensor out = result;
   if (result.numel() == 0) {
     result.resize_(self.sizes());
   }
+  if (!result.is_contiguous()) {
+    out = result.contiguous();
+  }
```

Comment:
Can you move lines 92 and 96-98 after line 101 which is when we do need to create the out tensor?